common/cache/lru_test.go

```diff
@@ -392,3 +420,116 @@ func TestCache_ItemHasCacheSizeDefined(t *testing.T) {
 
 	endWG.Wait()
 }
+
+func TestCache_ItemHasCacheSizeDefined_PutWithNewKeys(t *testing.T) {
```

Comment:
I added more tests below for PutWithNewKey/PutWithSameKey and PutIfNotExistWithNewKey/PutIfNotExistWithSameKey below